core/src/impl/Kokkos_Half_ReductionIdentity.hpp

```diff
@@ -0,0 +1,44 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef KOKKOS_HALF_REDUCTION_IDENTITY_HPP_
+#define KOKKOS_HALF_REDUCTION_IDENTITY_HPP_
+
+#include <Kokkos_ReductionIdentity.hpp>
+
+#if defined(KOKKOS_HALF_T_IS_FLOAT) && !KOKKOS_HALF_T_IS_FLOAT
+template <>
+struct Kokkos::reduction_identity<Kokkos::Experimental::half_t> {
+  static constexpr auto inf = Experimental::infinity_v<float>;
+  KOKKOS_FUNCTION static Experimental::half_t sum() noexcept { return 0; }
+  KOKKOS_FUNCTION static Experimental::half_t prod() noexcept { return 1; }
+  KOKKOS_FUNCTION static Experimental::half_t max() noexcept { return -inf; }
+  KOKKOS_FUNCTION static Experimental::half_t min() noexcept { return inf; }
```

Comment:
I'd rather not define another public variable and there doesn't seem much of a benefit to define it anyway instead of inlining it.
```suggestion
  KOKKOS_FUNCTION static Experimental::half_t sum() noexcept { return 0; }
  KOKKOS_FUNCTION static Experimental::half_t prod() noexcept { return 1; }
  KOKKOS_FUNCTION static Experimental::half_t max() noexcept { return -Experimental::infinity_v<float>; }
  KOKKOS_FUNCTION static Experimental::half_t min() noexcept { return Experimental::infinity_v<float>; }
```

Reply:
Furthermore, there isn't much of a reason to use the infinity value for `float`. We can either change it here, or in #8302. I don't think we need a third pull request.

Reply:
> Furthermore, there isn't much of a reason to use the infinity value for `float`. We can either change it here, or in #8302. I don't think we need a third pull request.

Using float avoids getting into the unary minus operator on `Impl::BitComparisonWrapper` question.
And no I do not agree to handle the finite math only issue here.

Reply:
Make inf a private symbol. Then I am good with this.